electron/src/preload/menu/preload-context.ts

```diff
@@ -17,83 +17,127 @@
-  ipcRenderer.send(EVENT_TYPE.ACTION.SAVE_PICTURE, new Uint8Array(bytes), timestamp);
+interface MenuItemTemplate {
+  label?: string;
+  enabled?: boolean;
+  type?: 'normal' | 'separator';
+  click?: () => void;
+}
+
+const performCut = async (): Promise<void> => {
+  try {
+    const selection = window.getSelection();
+    if (selection && selection.rangeCount > 0) {
+      const selectedText = selection.toString();
+      if (selectedText) {
+        await navigator.clipboard.writeText(selectedText);
+        selection.deleteFromDocument();
+      }
+    }
+  } catch (error) {
+    logger.warn('Cut operation failed:', error);
+  }
 };
 
-const copyPicture = async (url: RequestInfo): Promise<void> => {
-  const response = await fetch(url, {
-    headers: {
-      'User-Agent': config.userAgent,
-    },
-  });
-  const bytes = await response.arrayBuffer();
-  const image = nativeImage.createFromBuffer(Buffer.from(bytes));
-  clipboard.writeImage(image);
+const performCopy = async (): Promise<void> => {
+  try {
+    const selection = window.getSelection();
+    if (selection && selection.rangeCount > 0) {
+      const selectedText = selection.toString();
+      if (selectedText) {
+        await navigator.clipboard.writeText(selectedText);
+      }
+    }
+  } catch (error) {
+    logger.warn('Copy operation failed:', error);
+  }
 };
 
-const createDefaultMenu = (copyContext: string) =>
-  remote.Menu.buildFromTemplate([
-    {
-      click: () => clipboard.writeText(copyContext),
-      label: locale.getText('menuCopy'),
-    },
-  ]);
+const performPaste = async (): Promise<void> => {
+  try {
+    const text = await navigator.clipboard.readText();
+    const selection = window.getSelection();
+    if (selection && selection.rangeCount > 0) {
+      const range = selection.getRangeAt(0);
+      range.deleteContents();
+      range.insertNode(document.createTextNode(text));
+      range.collapse(false);
+    }
+  } catch (error) {
+    logger.warn('Paste operation failed:', error);
+  }
+};
+
+const performSelectAll = (): void => {
+  try {
+    const selection = window.getSelection();
+    if (selection) {
+      selection.selectAllChildren(document.body);
+    }
+  } catch (error) {
+    logger.warn('Select all operation failed:', error);
+  }
+};
 
-const createTextMenu = (params: ContextMenuParams, webContents: WebContents): ElectronMenu => {
-  const {editFlags, dictionarySuggestions} = params;
+const createTextMenu = (params: ContextMenuParams): MenuItemTemplate[] => {
+  const {editFlags = {}, dictionarySuggestions = []} = params;
 
-  const template: MenuItemConstructorOptions[] = [
+  const template: MenuItemTemplate[] = [
     {
-      click: (_menuItem, baseWindow) => sendToWebContents(baseWindow, EVENT_TYPE.EDIT.CUT),
-      enabled: editFlags.canCut,
       label: locale.getText('menuCut'),
+      enabled: editFlags.canCut,
+      click: () => {
+        performCut().catch(logger.error);
```

Comment:
performCut has already a try catch and never throws, .catch(logger.error) is unnecessary